web/modules/custom/dpl_campaign/src/Input/Facet.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Drupal\dpl_campaign\Input;
+
+/**
+ * A facet represents a named group of values which relate to a search result.
+ */
+class Facet {
+
+  /**
+   * The name of the facet.
+   *
+   * @var string
+   */
+  public string $name;
+
+  /**
+   * The values which relate to the search result.
+   *
+   * @var \Drupal\dpl_campaign\Input\Value[]
+   */
+  public array $values;
+
+  /**
+   * Facet constructor.
+   *
+   * @param string $name
+   *   The name of the facet.
+   * @param \Drupal\dpl_campaign\Input\Value[] $values
+   *   The values of the facet.
+   */
+  public function __construct(string $name, array $values) {
+    $this->name = $name;
```

Comment:
Why the difference in constructor style between? `Rule` and `Value` use PHP 8 style Constructor property promotion?

Reply:
PHPStan (or the version that we currently use) does not support adding types to arrays when using constructor property promotion 😕 .